src/api_client.rs

```diff
@@ -294,8 +295,13 @@ impl Client {
         unimplemented!("")
     }
 
-    pub async fn get_node_version() -> Result<String, Error> {
-        unimplemented!("")
+    pub async fn get_node_version(&self, endpoint: &str) -> Result<String, Error> {
+        let result: Result<VersionData, Error> = self.get(endpoint).await;
+
+        match result {
+            Result::Ok(res) => Ok(res.data["version"].to_string()),
+            Result::Err(error) => panic!("Problem making HTTP equest:\n {:?}", error),
+        }
```

Comment:
this is still replicating logic that has already been implemented in `self.get` if we leverage the type system a bit more.

the reason to do this is that we can avoid having to repeat this matching code you have here and also when the responses get more complex than just pulling out a `"version"` as a `String` we will want to leverage all of the `serde` code we have derived.
 

to move forward, 
instead of `VersionData` here, use the type `Value` from the `types` module and I think you'll see that you can get the "payload" of this response, then you will want to use the `VersionData`  you had here: https://github.com/jmcook1186/beacon-api-client/blob/15f23d83f4492101af9a22226a14d8f0767ead5a/src/types.rs#L11